### PR TITLE
Copyright offscreen text component

### LINKED ISSUE #483: Offscreen text for Image Copyright
**Is your feature request related to a problem? Please describe.**
We should apply offscreen text for the copyright of an image for screen-readers

**Describe the solution you'd like**
This offscreen text should be `Copyright  Getty  Images.`, where Getty Images is the value from the copyright field in the data feed. The screen-reader order should be `image -> copyright -> caption`.

We should use the visually hidden component.

**Testing**
- Use a screenreader to ensure that the copyright is read out, and is in the correct order
- Cypress test

**Additional Context**
- https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #483 

_This PR adds a new Copyright component homed within the Figure component. The Copyright is currently only offscreen and the copyrightHoler value is prepended with `Copyright `. Also if the copyrightHolder value is `BBC` we do not render anything. Following this PR it was suggested we refactor the Figure component into a container as the business logic for Caption and Copyright being conditional should not be handled in a presentational component._

- [x] Tests added for new features
- [x] Test engineer approval

### CODE REVIEW

File: src/app/components/Figure/Copyright/index.text.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '../../../helpers/tests/testHelpers';
+import VisuallyHiddenText from '../../VisuallyHiddenText';
+import Copyright from './index';
+
+describe('Copyright', () => {
```
[the remark this answers is not in the document]
Comment (pjlee11) [reply src/app/components/Figure/Copyright/index.text.jsx]: I disagree with CodeClimate on this duplication warning - it just the basic syntax of a Jest test with the first item being offscreen text

**Review by radiocontrolled — DISMISSED** [review]

File: src/app/components/Figure/Copyright/index.text.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (radiocontrolled) [reply src/app/components/Figure/Copyright/index.text.jsx]: Agree with you on this @pjlee11

File: src/app/containers/Image/index.jsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ const ImageContainer = ({ blocks }) => {
   const rawImageBlock = filterForBlockType(blocks, 'rawImage');
   const altTextBlock = filterForBlockType(blocks, 'altText');
   const captionBlock = filterForBlockType(blocks, 'caption');
+  const { copyrightHolder } = rawImageBlock.model;
```
Comment (jtart) [inline src/app/containers/Image/index.jsx]: This should be destructured on [line 29](https://github.com/bbc/simorgh/pull/515/files#diff-445cfd09f5ef4bfbe3544be7b7f2da00R29)

File: src/app/components/Figure/index.jsx [hunk 3]
```diff
@@ -1,9 +1,13 @@
 import React from 'react';
 import { string } from 'prop-types';
 import styled from 'styled-components';
+import Copyright from './Copyright';
```
Comment (jtart) [inline src/app/components/Figure/index.jsx]: Does this need to be it's own component? Feels a bit over-kill. Could it not just render the copyright into `VisuallyHiddenText`?

**Review by jtart — CHANGES_REQUESTED** [review]

File: src/app/components/Figure/index.jsx (hunk 3 above)
Comment (pjlee11) [reply src/app/components/Figure/index.jsx]: We could. However this way if we need to extend Copyright to be visual (not just offscreen) it doesn't need to be pulled out of Figure, also it means in the future the Figure presentational component doesn't need to worry about the translation for the offscreen text `Copyright `. 

This is likely mitigated if we have a Figure container component which I have suggested following this PR to implement the Copyright offscreen text.

Comment (pjlee11) [reply src/app/components/Figure/index.jsx]: I think personally the best solution would be to have a Figure container which handles the setting/translation of the offscreen text (`Copyright `) and then renders the VisuallyHiddenText therefore removing the need for the Copyright component and also not putting the offscreen text logic in a presentational component. That said I think that solution would be out of scope for the current ticket and should follow this PR

**Review by jtart — DISMISSED** [review]
I think this is fine for now but we should definitely look to re-factor our use of containers/components here, I think Sareh has something https://github.com/bbc/simorgh/pull/519

File: src/app/components/Figure/index.jsx [hunk 4]
```diff
@@ -2,8 +2,18 @@ import React from 'react';
...
 import Caption from './Caption';
+import VisuallyHiddenText from '../VisuallyHiddenText';
 import { GEL_SPACING_DBL } from '../../lib/constants/styles';
 
+const copyrightOffscreenText = 'Copyright ';
+
+const renderCopyright = copyright =>
+  copyright !== 'BBC' ? (
+    <VisuallyHiddenText>
+      {copyrightOffscreenText + copyright}
```
Comment (radiocontrolled) [inline src/app/components/Figure/index.jsx]: Minor change request: can you please use es6 templating here; then there also won't need to be a blank space on L8.

**Review by radiocontrolled — CHANGES_REQUESTED** [review]

**Review by radiocontrolled — APPROVED** [review]
👍

**Review by jtart — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM

**Pull request merged**